vpr/src/route/rr_graph_reader.cpp

```diff
@@ -701,7 +708,6 @@ void process_rr_node_indices(const DeviceGrid& grid) {
     /*Add the correct node into the vector
      * Note that CHANX and CHANY 's x and y are swapped due to the chan and seg convention.
```

Comment:
This reversing the order of the [ix][iy] indices for CHANY is pre-existing code, but odd.  Can leave, but worth considering if this should be fixed (simplified to always use [ix][iy]) in a later issue. Wouldn't change functionality, but likely less strange.

Reply:
Do you mean that the chan/seg convention should be changed to avoid the [ix][iy] swap? Or it should just be simplified for the `rr_node_indices` lookup?

Reply:
Thanks for the changes.

I looked into the rr-graph building code a bit more and see some other arrays with this [x][y] swap for y-directed routing elements. So lets leave it for now at least.  I find it a strange decision, but I'm not sure how much work it is to undo it, or whether it's worth it.  Orthogonal to all your changes so not gating on this commit.